Fix na_level bug

edelarua opened the pull request:
# Pull Request

<!--- Replace `#nnn` with your issue link for reference. -->

Fixes #nnn

edelarua commented:
> > Closes #911
> > I believe we should discuss refactoring this function entirely to process NA values in a more sensical way.
> 
> I think it is possible I added this. How do you think we should proceed?

I think we should remove the check for `NA` values - there's no reason why we shouldn't be able to process these in this/similar functions. Also, it wouldn't make a lot of sense for `na_level` to exist if we allowed `NA`s, since they would just show up in the table as `NA` and `na.rm` would still work - if the user wants to explicitly define/remove the `NA` level in pre-processing, they are still free to do so.

Melkiades commented:
> > > Closes #911
> > > I believe we should discuss refactoring this function entirely to process NA values in a more sensical way.
> > 
> > 
> > I think it is possible I added this. How do you think we should proceed?
> 
> I think we should remove the check for `NA` values - there's no reason why we shouldn't be able to process these in this/similar functions. Also, it wouldn't make a lot of sense for `na_level` to exist if we allowed `NA`s, since they would just show up in the table as `NA` and `na.rm` would still work - if the user wants to explicitly define/remove the `NA` level in pre-processing, they are still free to do so.

It makes sense. But what if we want just to replace output NAs with custom strings? I think that should remain, i.e. when we need to have "NE" instead. If there is another way than using `na_str`, I apologize for the silly question!

edelarua commented:
> > > > Closes #911
> > > > I believe we should discuss refactoring this function entirely to process NA values in a more sensical way.
> > > 
> > > 
> > > I think it is possible I added this. How do you think we should proceed?
> > 
> > 
> > I think we should remove the check for `NA` values - there's no reason why we shouldn't be able to process these in this/similar functions. Also, it wouldn't make a lot of sense for `na_level` to exist if we allowed `NA`s, since they would just show up in the table as `NA` and `na.rm` would still work - if the user wants to explicitly define/remove the `NA` level in pre-processing, they are still free to do so.
> 
> It makes sense. But what if we want just to replace output NAs with custom strings? I think that should remain, i.e. when we need to have "NE" instead. If there is another way than using `na_str`, I apologize for the silly question!

That does make sense and I think we should add that as well, but that's not the current function of the `na_level` argument in this function - currently, since `NA` is not allowed in the table, users can specify `na_level`, for example `na_level = "<Missing>"`, and this value will be treated as `NA` (i.e. removed by `na.rm`).

Melkiades reviewed (approved):
Lgtm! We can discuss later for the `na_str` propagation ;)
